scripts/TunnelProxy.ps1

```diff
@@ -1,89 +1,64 @@
             #htpasswd -bc $passwordsFile $ProxyCredential.Username $ProxyCredential.GetNetworkCredential().Password
             $basicAuthConfig = Get-Content $basicAuthFile -Raw
-        } else {
+        }
+        else {
             $basicAuthConfig = ""
         }
 
-        if ($UseInspection) {
+        if ($Context.UseInspection) {
             Write-Header "Setting up proxy for TLS inspection..."
-            (Get-Content $ExcludeDomainFile) -replace "##DOMAIN_NAME##", "$TunnelServer" | out-file (Join-Path $pwd -ChildPath "proxy" -AdditionalChildPath "ssl_exclude_domains.tmp")
+            (Get-Content $ExcludeDomainFile) -replace "##DOMAIN_NAME##", "$Context.TunnelFQDN" | out-file (Join-Path $pwd -ChildPath "proxy" -AdditionalChildPath "ssl_exclude_domains.tmp")
             $breakAndInspectConfig = Join-Path $pwd -ChildPath "proxy" -AdditionalChildPath "BreakAndInspect.conf"
             $breakAndInspectConfig = Get-Content $breakAndInspectConfig -Raw
-        } else {
+        }
+        else {
             $breakAndInspectConfig = Join-Path $pwd -ChildPath "proxy" -AdditionalChildPath "vanilla.conf"
             $breakAndInspectConfig = Get-Content $breakAndInspectConfig -Raw
         }
 
-        if ($UseAllowList) {
+        if ($Context.UseAllowList) {
             $allowlistConfig = Join-Path $pwd -ChildPath "proxy" -AdditionalChildPath "allowlist.conf"
             $allowlistConfig = Get-Content $allowlistConfig -Raw
-        } else {
+        }
+        else {
             $allowlistConfig = ""
         }
 
         $allowlistFile = Join-Path $pwd -ChildPath "proxy" -AdditionalChildPath "allowlist"
         $proxyScript = Join-Path $pwd -ChildPath "scripts" -AdditionalChildPath "proxySetup.sh"
         $pacFile = Join-Path $pwd -ChildPath "nginx_data" -AdditionalChildPath "tunnel.pac"
 
-        (Get-Content $configFile) -replace "##DOMAIN_NAME##", "$TunnelServer" -replace "##BASIC_AUTH##", "$basicAuthConfig" -replace "##BREAK_AND_INSPECT##", "$breakAndInspectConfig" -replace "##ALLOWLIST##", "$allowlistConfig" | out-file "$configFile.tmp"
-        (Get-Content $allowlistFile) -replace "##DOMAIN_NAME##", "$TunnelServer" | out-file "$allowlistFile.tmp"
+        (Get-Content $configFile) -replace "##DOMAIN_NAME##", "$($Context.TunnelFQDN)" -replace "##BASIC_AUTH##", "$basicAuthConfig" -replace "##BREAK_AND_INSPECT##", "$breakAndInspectConfig" -replace "##ALLOWLIST##", "$allowlistConfig" | out-file "$configFile.tmp"
+        (Get-Content $allowlistFile) -replace "##DOMAIN_NAME##", "$($Context.TunnelFQDN)" | out-file "$allowlistFile.tmp"
 
-        $ProxyURL = "proxy.$TunnelServer"
+        $ProxyURL = "proxy.$($Context.TunnelFQDN)"
         (Get-Content $pacFile) -replace "##PROXY_URL##", "$ProxyURL" | out-file "$pacFile.tmp"
         $pacFile = "$pacFile.tmp"
 
-        $proxyBypassNames = ("www.google.com", "excluded.$($TunnelServer)")
+        $proxyBypassNames = ("www.google.com", "excluded.$($Context.TunnelFQDN)")
```

Comment:
We should probably move things like "www.google.com" to the context intialization so we have all defaults in one spot